clang/include/clang/Frontend/CompilerInstance.h

```diff
@@ -644,6 +648,12 @@ class CompilerInstance : public ModuleLoader {
   /// the compiler instance takes ownership of \p Value.
   void setCodeCompletionConsumer(CodeCompleteConsumer *Value);
 
+  /// }
+  /// @name Back-end Pass Plugins
+  /// @{
+
+  std::vector<llvm::PassPlugin *> &getPassPlugins() { return PassPlugins; }
```

Comment:
```suggestion
  llvm::ArrayRef<llvm::PassPlugin*> getPassPlugins() { return PassPlugins; }
```

This may be better than returning a mutable reference to an `std::vector`.